power_dashboard/app.py

```diff
@@ -339,3 +342,38 @@ def load_forecast_model():
         ax.set_ylabel("Carbon Intensity (gCO2eq/kWh)")
         fig.autofmt_xdate()
         st.pyplot(fig)
+
+    with tab3:
+        st.title("Personal CO2 Calculator from Green Button XML File")
+
+        uploaded_file = st.file_uploader("Choose a Green Button XML file")
```

Comment:
I tested the uploader on my machine and it successfully displayed the table.  I'm taking this is still in a draft state and you're working on combining it with the CO2 intensity history to produce a footprint?

Reply:
Oh, great! Thank you. Yes, this is a draft. I am working on the CO history now